ChronoKeeper/KeeperRegistry.h

```diff
@@ -0,0 +1,82 @@
+#ifndef KEEPER_REGISTRY_H
+#define KEEPER_REGISTRY_H
+
+#include <mutex>
+#include <vector>
+#include <map>
+
+#include "KeeperIdCard.h"
+#include "KeeperStatsMsg.h"
+
+
+namespace chronolog
+{
+
+
+class KeeperRegistry
+{
+public:
+
+struct KeeperProcessEntry
+{
+	KeeperProcessEntry(KeeperIdCard const& keeper_id_card)
+		: idCard(keeper_id_card)
+		, lastStatsTime(0)
+		, activeStoryCount(0)
+	{}
+
+	KeeperIdCard idCard;
+	uint64_t	lastStatsTime;
+	uint32_t	activeStoryCount;
+};
+
+	KeeperRegistry(){}
+	~KeeperRegistry()=default;
+
+	int registerKeeperProcess( KeeperIdCard const& keeper_id_card)
+	{ 
+	   std::lock_guard<std::mutex> lock_guard(registryLock);
```

Comment:
Why do you have a lock here and not in the other functions?

Reply:
all 3 implemented Registry functions do have locks .   updateStats is not yet implemented